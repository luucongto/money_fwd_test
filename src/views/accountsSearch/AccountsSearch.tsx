import {
  CButton,
  CCard,
  CCardBody, CCol, CInput, CRow
} from '@coreui/react'
import { useState } from 'react'

import { useHistory } from 'react-router-dom'
import { useAppDispatch } from '../../redux/hooks'
import { actions } from '../../redux/reducers/user'
const AccountsSearch = (): JSX.Element => {
  const [userId, setUserId] = useState('')
  const history = useHistory()
  const handleSubmit = (): void => {
    history.push(`/accountDisplay/${userId}`)
  }

  return (
    <>
      <CCard>
        <CCardBody>
          <CRow>
            <CCol sm='5'>
              <CInput
                id='name' placeholder='Enter the userID' required value={userId} onChange={(event: React.ChangeEvent<HTMLInputElement>) => {
                  setUserId(event.target.value)
                }}
                onKeyDown={(event) => {
                  if (event.key === 'Enter') {
                    handleSubmit()
                  }
                }}
              />
            </CCol>
            <CCol>
              <CButton type='submit' color='primary' style={{ justifySelf: 'center' }} onClick={handleSubmit}>Search</CButton>
            </CCol>
          </CRow>
        </CCardBody>
      </CCard>
    </>
  )
}

export default AccountsSearch
