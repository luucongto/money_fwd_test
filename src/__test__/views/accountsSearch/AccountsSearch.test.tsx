import React from 'react'
import AccountsSearch from '../../../views/accountsSearch/AccountsSearch'
import { render, screen } from '@testing-library/react'
import userEvent from '@testing-library/user-event'
import { shallow } from 'enzyme'
import { Provider } from 'react-redux'
import configureStore from 'redux-mock-store'
// Before your test, do next setting
const middlewares = [] // If you are not using any middleware, leave the array empty or import it and put it in
const mockStore = configureStore(middlewares) // Setting of store based on the middlewares.
const initState = {
  sidebar: {
    sidebarShow: 'responsive'
  }
}
// Create the store to pass as prop in the <Provider>
const store = mockStore(initState)

jest.mock('react-router-dom', () => ({
  ...jest.requireActual('react-router-dom'),
  useHistory: () => ({
    push: jest.fn()
  })
}))

test('renders AccountsSearch without crashing', () => {
  const wrapperResult = shallow(
    <Provider store={store}>
      <AccountsSearch />
    </Provider>
  )
  expect(wrapperResult).toMatchSnapshot()
})

test('render AccountsSearch has Search Button', () => {
  render(
    <Provider store={store}>
      <AccountsSearch />
    </Provider>
  )
  const search = screen.getByText(/Search/i)
  expect(search).toBeInTheDocument()
  userEvent.click(search)
})
describe('AccountsSearch input', () => {
  let input: Element
  beforeEach(() => {
    render(
      <Provider store={store}>
        <AccountsSearch />
      </Provider>
    )
    input = screen.getByPlaceholderText(/Enter the userID/i)
  })

  test('render AccountsSearch has input', () => {
    expect(input).toBeInTheDocument()
    userEvent.type(input, '1')
  })

  test('render AccountsSearch has input with enter key', () => {
    userEvent.type(input, '1{enter}')
  })
})
