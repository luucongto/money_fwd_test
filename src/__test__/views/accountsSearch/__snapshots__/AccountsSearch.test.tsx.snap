// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders AccountsSearch without crashing 1`] = `ShallowWrapper {}`;
